conngen/conngen.cpp

```diff
@@ -132,86 +50,278 @@ nest::cg_connect( nest::ConnectionGeneratorDatum& cg,
     throw UnknownSynapseType( synmodel_name.toString() );
   const index synmodel_id = static_cast< index >( synmodel );
 
-  RangeSet source_ranges;
-  cg_get_ranges( source_ranges, ( *sources ) );
+  RangeSet source_ranges, target_ranges;
 
-  RangeSet target_ranges;
-  cg_get_ranges( target_ranges, ( *targets ) );
+  cg_get_ranges( source_ranges, source_gids );
+  cg_get_ranges( target_ranges, target_gids );
+  cg_set_masks( cg, source_ranges, target_ranges );
```

Comment:
Here and in a few other places, we always have two calls to `cg_get_ranges()` followed by `cg_set_masks()`. `cg_get_ranges()` does not seem to be used in any other context. Wouldn't it be easier (and make for more readable code) to pass `source_gids` and `target_gids` directly to `cg_set_masks()` and then let `cg_set_masks()` get the ranges. 

One thing I really dislike about C++ is that in the `cg_get_ranges()` calls, there is no clear, visual data flow.
```C++
source_ranges = cg_get_ranges( source_gids );
```
would look so much cleaner, but then one would need smart pointers ...